### ISSUE TITLE
[BUG] mse_loss_backward: reads grad_output out of bounds for reduction='mean'/'sum', returning wrong gradients

### ISSUE THREAD

**Issue by truong-v** [FlagOpen/FlagGems#5212]
## Summary

For `reduction='mean'` and `'sum'`, the `mse_loss` forward returns a scalar, so autograd hands the backward a 0-dim `grad_output`, which ATen broadcasts against `self`. `mse_loss_backward_kernel` instead loads `grad_output[offset]` for every output element, and the wrapper only calls `grad_output.flatten()` (0-dim → one element). The kernel therefore reads `numel - 1` elements past a one-element buffer: gradient element 0 is correct, every other element is scaled by whatever memory follows the allocation. No error is raised — and `'mean'` is the default reduction of `F.mse_loss`, so a plain `F.mse_loss(pred, target).backward()` under `use_gems()` silently produces wrong gradients.

`tests/test_mse_loss_backward.py` passes a full-shape `grad_output` for all three reductions; the broadcast is the identity there, which is why the suite never sees it.

The kernel was added in #3675.

## Reproduction

```python
import torch
import torch.nn.functional as F
import flag_gems

torch.manual_seed(0)
x = torch.randn(4, device="cuda", requires_grad=True)
target = torch.randn(4, device="cuda")

F.mse_loss(x, target).backward()
ref = x.grad.clone()
x.grad = None

with flag_gems.use_gems():
    F.mse_loss(x, target).backward()

print("eager     :", ref.tolist())
print("flag_gems :", x.grad.tolist())
```

```
eager     : [-0.5527, 0.0635, -1.7838, 0.4401]
flag_gems : [-0.5527, 0.2142, -22.7049, 0.3409]
```

Only element 0 matches; the wrong values depend on allocator state because they come from adjacent memory.

Reproduced at master (75cf3ea) on L40S and H100, torch 2.6.0+cu124, triton 3.2.0.
